frog/imports/client/Wiki/ModalPageSettings.js

```diff
@@ -81,72 +86,79 @@ class NewPageModal extends React.Component<PropsT, StateT> {
     super(props);
     this.state = {
       currentTab: 0,
-      pageTitle: '',
+      pageTitle: this.props.title || '',
       pageTitleValid: true,
       open: true,
       expanded: false,
-      socialPlane: 3,
-      allowView: true,
-      allowEdit: true
+      socialPlane: this.props.socialPlane || 3,
+      pageSettings: this.props.pageSettings || {
+        allowView: true,
+        allowEdit: true,
+        readOnly: false,
+        hidden: false
+      },
+      error: null
     };
   }
 
   handleTitleChange = (e: any) => {
-    this.handleErrorClearing(e.target.value);
-    this.setState({ pageTitle: e.target.value });
+    this.setState({ pageTitle: e.target.value, error: null });
   };
 
   handleTabs = (e: any, value: number) => {
     this.setState({ currentTab: value });
   };
 
   handleSocialPlaneChange = (e: any) => {
-    if (e.target.value === 'everyone') {
+    if (e.target.value === 3) {
       this.setState({
-        socialPlane: e.target.value,
-        allowView: true,
-        allowEdit: true
+        socialPlane: 3,
+        pageSettings: {
+          ...this.state.pageSettings,
+          allowView: true,
+          allowEdit: true
+        }
       });
     } else {
       this.setState({ socialPlane: e.target.value });
     }
   };
 
-  handleChangeAllowView = () => {
-    this.setState({ allowView: !this.state.allowView });
-  };
-
-  handleChangeAllowEdit = () => {
-    this.setState({ allowEdit: !this.state.allowEdit });
-  };
+  handleCheckbox = target => e =>
+    this.setState({
+      ...this.state,
+      pageSettings: { ...this.state.pageSettings, [target]: e.target.checked }
+    });
 
   handleConfig = conf => {
     this.setState({ activityConfig: conf });
   };
 
-  handleCreate = () => {
+  handleSubmit = () => {
     const {
       pageTitle,
       socialPlane,
       activityConfig,
-      operatorConfig
+      operatorConfig,
+      pageSettings
     } = this.state;
-    this.props.onCreate(pageTitle, socialPlane, activityConfig, operatorConfig);
+    this.props
+      .onSubmit(
+        pageTitle,
+        socialPlane,
+        activityConfig,
+        operatorConfig,
+        pageSettings
+      )
+      .then(error => {
```

Comment:
Do we expect props.onSubmit to return a Promise? If so, specify that in the type. If the parent does not respect this invariant, this will lead to an exception